libsolidity/ast/Types.cpp

```diff
@@ -483,8 +555,6 @@ bool IntegerType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 		IntegerType const& convertTo = dynamic_cast<IntegerType const&>(_convertTo);
 		if (convertTo.m_bits < m_bits)
 			return false;
-		if (isAddress())
```

Comment:
I think this was missing an ``else if (convertTo.isAddress()) return isAddress();`` before, resulting in the ``commonType`` of addresses and integers to be the address type and thus allowing int->address casts in ternary expressions.

Reply:
Did we introduce a test for this assumption?